psutil/_psutil_windows.c

```diff
@@ -363,7 +363,30 @@ psutil_proc_exe(PyObject *self, PyObject *args) {
         sizeof(SYSTEM_PROCESS_ID_INFORMATION),
         NULL);
 
-    if (status == STATUS_INFO_LENGTH_MISMATCH) {
+    if (status == STATUS_INFO_LENGTH_MISMATCH && processIdInfo.ImageName.MaximumLength <= bufferSize) {
+        // Required length was NOT stored in MaximumLength (WOW64 issue).
+        
+        ULONG maxBufferSize = 0x7FFF * 2; //NTFS_MAX_PATH * sizeof(wchar_t)
+        
+        do {
+            // Iteratively double the size of the buffer up to maxBufferSize
+            bufferSize *= 2;
+            
+            FREE(buffer);
```

Comment:
If `buffer` was never initialized, isn't this going to be a problem? Perhaps it makes sense to define buffer as:

```
PVOID buffer = NULL;
```

...and in here do:

```
if (buffer != NULL)
      FREE(buffer);
```

(not tested)